espnet2/bin/s2t_inference.py

```diff
@@ -478,11 +504,14 @@ def _decode_single_sample(self, enc: torch.Tensor):
 
             # remove sos/eos and get results
             last_pos = -1
+            start_pos = 1 if self.partial_ar else 0
             if isinstance(hyp.yseq, list):
-                token_int = hyp.yseq[:last_pos]
+                token_int = hyp.yseq[start_pos:last_pos]
             else:
-                token_int = hyp.yseq[:last_pos].tolist()
-            token_int = token_int[token_int.index(self.s2t_model.sos) + 1 :]
+                token_int = hyp.yseq[start_pos:last_pos].tolist()
+
+            if not self.partial_ar:
+                token_int = token_int[token_int.index(self.s2t_model.sos) + 1 :]
```

Comment:
There might be an extra text prompt before `sos`. Is this supported by the partial AR inference?

Reply:
Am I right that those prompts are not generated by the decoder? If so, there will be no extra prompts generated by the `PartiallyARInference` class.
If extra prompts would be generated by decoder then we need some fix.

Reply:
They are not generated by the decoder. They are provided as the `hyp_primer` in the current beam search, just like `sos`.
Then I guess this prompt is not supported in the PartiallyAR interface

Reply:
Thank you, then we do not need to fix this part, as the output from the beam search class does not contain the extra prompts.
This `token_int` is the same format as Mask-CTC model, so it has mask token at the beginning and end.